File: gaming - copia 2/src/main/java/com/oruga/backend/model/Reporte.java
package com.oruga.backend.model;

import com.oruga.backend.model.enums.EstadoReporte;
import com.oruga.backend.model.enums.TipoReporte;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

/**
 * Entidad Reporte - Replica del modelo Android
 * Modelo para reportes de contenido inapropiado
 */
@Entity
@Table(name = "reportes")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Reporte {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Usuario que reporta
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reportado_por_id", nullable = false)
    private Usuario reportadoPor;

    @Column(nullable = false)
    private String reportadoPorNombre;

    // Tipo de reporte
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoReporte tipoReporte;

    @Column(nullable = false)
    private Integer idContenido; // ID de publicación o mensaje

    // Detalles
    @Column(nullable = false)
    private String razon;

    @Column(length = 2000)
    private String descripcion = "";

    // Estado
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EstadoReporte estado = EstadoReporte.PENDIENTE;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime fechaReporte;

    // Resolución (por moderador)
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "moderador_id")
    private Usuario moderador;

    @Column
    private String moderadorNombre;

    @Column
    private LocalDateTime fechaResolucion;

    @Column(length = 2000)
    private String accionTomada = "";
}

File: MyApplication3/app/src/main/java/com/example/myapplication2/data/models/Reporte.kt
package com.example.myapplication.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Modelo para reportes de contenido inapropiado
 */
@Entity(tableName = "reportes")
data class Reporte(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,

    // Usuario que reporta
    val reportadoPorId: Int,
    val reportadoPorNombre: String,

    // Tipo de reporte
    val tipoReporte: TipoReporte,
    val idContenido: Int, // ID de publicación o mensaje

    // Detalles
    val razon: String,
    val descripcion: String = "",

    // Estado
    val estado: EstadoReporte = EstadoReporte.PENDIENTE,
    val fechaReporte: Long = System.currentTimeMillis(),

    // Resolución (por moderador)
    val moderadorId: Int? = null,
    val moderadorNombre: String? = null,
    val fechaResolucion: Long? = null,
    val accionTomada: String = ""
)

/**
 * Tipos de reporte
 */
enum class TipoReporte {
    PUBLICACION,
    MENSAJE,
    USUARIO,
    OTRO
}

/**
 * Estados de reporte
 */
enum class EstadoReporte {
    PENDIENTE,
    EN_REVISION,
    RESUELTO,
    RECHAZADO
}